core/sections/clause_8_ogc-process-description.adoc

```diff
@@ -157,6 +157,26 @@ The following JSON Schema fragment illustrates the use of the `format` key to in
 }
 ----
 
+==== Data classes
+
+One common input type that a process might accept is a https://docs.ogc.org/is/17-069r4/17-069r4.html#_collection_[feature collection] indicating that the process will somehow operate over the items of the collection.  This implies that the process will have certain expections about the structure of the collection with regard to which properties the collection contains, their types, etc.  In order to properly handle any arbitrary input collection a process would need to inspect the structure of the collection to ensure that all the expected properties are present with the expected types, etc.  To alleviate the server from having to perform such a tedious, and perhaps computationally expensive, schema validation step this Standard introduces the concept of the _Data Class_ via the `dataClasses` parameter.  
+
+The value of the `dataClasses` parameter is an array of URIs.  Each URI identifies a predefined set of properties or a sub-schema.  Two data resources tagged with the same data class URI value can be assumed to each contain all the properties defined by the data class.  This equivalence allows a server to quickly validate that an input data resource meets the server's expections in terms of the properties available for processing, their types, etc. simply by comparing data class URI values.  If the data class URI of the input data resource matches one of the data class URIs specified in the description of the process input (via the `dataClasses` parameter) then the server can be assured that the process can operate on that data resource.
+
+The `dataClasses` parameter is an array allowing process inputs to be described that can handle a variety of data classes.  As long as the data class URI associated with an input data resource matches at least one of the data class URIs listed in the `dataClasses` array the server can assume that the process can operate on that input data process.
```

Comment:
@fmigneault Excellent catch.

In the context of a GeoData (collection), the array means that it conforms to all.

In the context of a Process or Style, I think it would need to mean "any" as you suggest, or we revert to a single class.

> Should that be an entirely new dataClass definition on its own that would semantically duplicate the conditions of both "child" dataClasses?

Yes (I would see these as "ancestor" dataClasses in terms of OOP hierarchy / Multiple Inheritance)

Reply:
OK. So, similar to the `conformsTo` of servers? The collection exposes that it conforms to a set of definitions, but the process can work with any (subset) of those definitions. Seems like a fair assumption.

However, that is maybe problematic if a process must use (dataClassA, dataClassB) simultaneously. If we define a new dataClassC with ancestors (dataClassA, dataClassB) to address this situation, then every collection out there must be updated with (dataClassA, dataClassB, dataClassC) for this process to be able to access it as dataClassC. I don't think that scales well. The process would be unusable until all desired collections are updated with dataClassC, or it needs to simply ignores the `dataClasses` requirement (kind of defeating the purpose).

Reply:
> So, similar to the conformsTo of servers? 

Exactly.

> However, that is maybe problematic if a process must use (dataClassA, dataClassB) simultaneously. 

My thinking is that you would not often have a case of orthogonal conformance to different classes, but rather more often collections to specify conformance to a basic thing like "geometry is polygon/multipolygon" and something more specific like "elevation contour polygons with `elevation` property in meters", in addition to the basic thing.

And the most important use case is really a specific schema that a data publisher defines, where any dataset organized exactly the same way (whether published by that same publisher or not) would also conform.

Reply:
@fmigneault that is correct.  The description currently says "... matches one of ...".  I've reworded it to say "...matches at least one of ..." to handle the case you describe.  Please verify if this is OK.